static/src/javascripts/projects/common/modules/ui/autoupdate.js

```diff
@@ -0,0 +1,242 @@
+import NotificationCounter from 'common/modules/ui/notification-counter';
+import { checkElemsForVideos } from 'common/modules/atoms/youtube';
+
+type autoUpdateOptions = {
+    toastOffsetTop: number,
+    minUpdateDelay: number,
+    maxUpdateDelay: number,
+    backoffMultiplier: number,
+};
+
+const autoUpdate = (opts?: autoUpdateOptions) => {
+    const options = Object.assign(
+        {
+            toastOffsetTop: 12,
+            minUpdateDelay:
+                (detect.isBreakpoint({ min: 'desktop' }) ? 10 : 30) * 1000,
+            maxUpdateDelay: 20 * 60 * 1000, // 20 mins
+            backoffMultiplier: 0.75, // increase or decrease the back off rate by modifying this
+        },
+        opts
+    );
+
+    // Cache selectors
+    const $liveblogBody = $('.js-liveblog-body');
+    const $toastButton = $('.toast__button');
+    const $toastText = $('.toast__text', $toastButton);
+    const toastContainer = qwery('.toast__container')[0];
+    let currentUpdateDelay = options.minUpdateDelay;
+    let latestBlockId = $liveblogBody.data('most-recent-block');
+    let unreadBlocksNo = 0;
+    let updateTimeoutId;
+
+    const updateDelay = (delay: number): void => {
+        let newDelay;
+
+        if (detect.pageVisible()) {
+            newDelay = options.minUpdateDelay;
+        } else {
+            newDelay = Math.min(delay * 1.5, options.maxUpdateDelay);
+        }
+
+        currentUpdateDelay = newDelay;
+    };
+
+    const scrolledPastTopBlock = (): boolean =>
+        $liveblogBody.offset().top < window.pageYOffset;
+
+    const isLivePage = !window.location.search.includes('?page=');
+
+    const revealInjectedElements = (): void => {
+        fastdom.write(() => {
+            $('.autoupdate--hidden', $liveblogBody)
+                .addClass('autoupdate--highlight')
+                .removeClass('autoupdate--hidden');
+            mediator.emit('modules:autoupdate:unread', 0);
+        });
+    };
+
+    const toastButtonRefresh = (): void => {
+        fastdom.write(() => {
+            if (unreadBlocksNo > 0) {
+                const updateText = unreadBlocksNo > 1
+                    ? ' new updates'
+                    : ' new update';
+                $toastButton.removeClass('toast__button--closed');
+                $(toastContainer).addClass('toast__container--open');
+                $toastText.html(unreadBlocksNo + updateText);
+            } else {
+                $toastButton
+                    .removeClass('loading')
+                    .addClass('toast__button--closed');
+                $(toastContainer).removeClass('toast__container--open');
+            }
+        });
+    };
+
+    const injectNewBlocks = (newBlocks: string): void => {
+        // Clean up blocks before insertion
+        const resultHtml = $.create(`<div>${newBlocks}</div>`)[0];
+        let elementsToAdd;
+
+        fastdom.write(() => {
+            bonzo(resultHtml.children).addClass('autoupdate--hidden');
```

Comment:
Could we potentially use `$` here instead of `bonzo` and drop `bonzo` from this module?